sycl/doc/extensions/SYCL_pipes_lowering_to_SPIRV.rst

```diff
@@ -0,0 +1,111 @@
+SYCL INTEL spatial pipes
+========================
+
+Introduction
+============
+
+SPIR-V is first class target in which SYCL pipes should be representable, and
+pipes are already exposed within SPIR-V. This document describes how SYCL pipes
+are being lowered to SPIR-V.
+
+OpenCL 2.2 program pipe representation in SPIR-V
+================================================
+
+The SPIR-V program pipe representation is leveraged, to be an underlying
+representation of intra-kernel and inter-kernel static pipe connectivity.
+The SPIR-V pipe representation exists in a series of pieces:
+
+ - OpTypePipeStorage: Type representing memory allocated for storage of data
+ within a pipe. Used for OpenCL 2.2 program pipes (program-scope pipes) that
+ the host program is not aware of, but that enables connectivity between
+ kernels.
+
+ - OpConstantPipeStorage: Instruction that creates an OpTypePipeStorage object.
+ Requires packet size (number of bytes) and capacity (number of packets) to be
+ defined.
+
+ - OpTypePipe: A pipe object that can act as a read/write endpoint of some pipe
+ storage, either allocated by a host and passed as a kernel argument, or
+ allocated at "program scope" through a pipe storage object.
+
+ - OpCreatePipeFromPipeStorage: Creates pipe object (that can be read/written)
+ from an OpTypePipeStorage instance.
+
+ - OpReadPipe / OpWritePipe: Read packet from or write packet to a pipe object.
+
+Lowering of kernel to kernel pipes to SPIR-V (non-blocking)
+===========================================================
+
+This connectivity is achieved through OpTypePipeStorage, which allows a SPIR-V
+device consumer to leverage static connectivity. An OpConstantPipeStorage
+instruction must create a single instance of OpPipeStorage for each kernel to
+kernel pipe type used by any kernel within the application.
+
+OpTypePipe objects is created from OpPipeStorage using
+OpCreatePipeFromPipeStorage. The number of OpTypePipe objects created from an
+OpPipeStorage object is an implementation detail, as are the access qualifiers
+applied to those types. For example, an implementation is free to create a
+different OpTypePipe corresponding to each read and write, with unidirectional
+access qualifiers annotated, or it can create fewer OpTypePipe objects, although
+read and write pipes must be distinct according to OpReadPipe and OpWritePipe
+rules.
+
+NOTE: The SPIR-V OpReadPipe and OpWritePipe instructions are non-blocking.
+
+Details SPIR-V representation in LLVM IR
+========================================
+
+Pipe built-ins are mangled in LLVM IR to make it SPIR-V friendly.
+As an example:
+ OpReadPipe                  -> __spirv_ReadPipe
+ OpWritePipe                 -> __spirv_WritePipe
+ OpCreatePipeFromPipeStorage -> __spirv_CreatePipeFromPipeStorage_{read|write}
+
+where 0 and 1 are representing read_only and write_only access qualifiers.
```

Comment:
Sorry, there were strings with pipe types representation in SPIR-V and SPIR-V friendly IR (since I reuse OpenCL pipes).
Was something like that:
read pipe: __spirv_Pipe_0
write pipe: __spirv_Pipe_1